R/shinyform.R

```diff
@@ -73,15 +117,65 @@ loadDataFlatfile <- function(storage) {
   data
 }
 
+
+
+# Takes data from your shinyforms inputs and saves it to a google doc file
+# @param data Dataframe taken from input shiny object
+# @param storage A list with variable type defining users perferred type of storage and storage key
 saveDataGsheets <- function(data, storage) {
   gs_add_row(gs_key(storage$key), input = data)
 }
+
+
+
+# Takes data from a google doc file and passes it to your shiny app.
+# @param storage A list with variable type defining users perferred type of storage and storage key
 loadDataGsheets <- function() {
   gs_read_csv(gs_key(storage$key))
 }
 
+
+
+#' Creates the UI form component for shinyforms. 
+#'
+#' @param formInfo A list with param: id, questions and storage 
+#' 
+#' @examples  
+#' if (interactive()) {
+#' library(shiny)
+#' library(shinyforms)
+#'
+#' questions <- list(
+#'   list(id = "name", type = "text", title = "Name", mandatory = TRUE),
+#'   list(id = "age", type = "numeric", title = "Age"),
+#'   list(id = "favourite_pkg", type = "text", title = "Favourite R package"),
+#'   list(id = "terms", type = "checkbox", title = "I agree to the terms")
+#' )
+#' formInfo <- list( 
+#' id = "basicinfo",
+#' questions = questions,
+#' storage = list(
+#'   # Right now, only flat file storage is supported
+#'   type = STORAGE_TYPES$FLATFILE,
+#'   # The path where responses are stored
+#'   path = "responses"
+#' )
+#' )
+#' ui <- fluidPage(
+#'   formUI(formInfo)
+#' )
+#'
+#' server <- function(input, output, session) {
+#'   formServer(formInfo)
+#' }
+#'
+#' shinyApp(ui = ui, server = server)
+#'}
 #' @export
 formUI <- function(formInfo) {
+  if (!testList(formInfo)) {
+    stop("`formInfo` missing from shinyApp")
```

Comment:
This error message is incorrect. It should say "formInfo is not a valid list"

Reply:
updated all stop messages to read "formInfo is not a valid list"- will be in the next commit